lib/active_merchant/billing/gateways/safe_charge.rb

```diff
@@ -154,6 +169,15 @@ def add_customer_details(post, payment, options)
         post[:sg_Email] = options[:email]
       end
 
+      def add_external_mpi_data(post, options)
+        post[:sg_eci] = options[:three_d_secure][:eci] if options[:three_d_secure][:eci]
+        post[:sg_cavv] = options[:three_d_secure][:cavv] if options[:three_d_secure][:cavv]
+        post[:sg_dsTransID] = options[:three_d_secure][:ds_transaction_id] if options[:three_d_secure][:ds_transaction_id]
+        post[:sg_threeDSProtocolVersion] = options[:three_d_secure][:version] || options[:three_d_secure][:ds_transaction_id] ? '2' : '1'
```

Comment:
This line is a bit confusing to me - I am unsure in Ruby whether the `||` operator or the `?` operator is applied first. Could you consider adding parentheses to make it obvious how the operation is grouped?